src/clj/puppetlabs/http/client/async.clj

```diff
@@ -242,7 +152,7 @@
    * :query-params - used to set the query parameters of an http request"
   [opts :- common/RawUserRequestOptions
    callback :- common/ResponseCallbackFn
-   client :- HttpAsyncClient]
+   client :- common/Client]
```

Comment:
This seems to be a synonym for ```CloseableHttpAsyncClient```. There's a comment in there that suggests replacing with a protocol. I guess we won't ever do that. I wonder if we can just get rid of ```common/client``` altogether.